### PR TITLE
feat: upgrade seal to v3.5.5

### PR DESCRIPTION
**Pull request by s0l0ist**
## Description

Updating to latest SEAL v3.5.5 for performance enhancements.

Cleaned up github workflows, updated to latest Ubuntu LTS (20.04) which uses a more modern version of GCC (9) which is able to properly build the most recent version of Microsoft SEAL.

## Affected Dependencies

- SEAL
- gcc >= 9

## How has this been tested?

- Build tests/benchmark passing

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -6,32 +6,30 @@ on:
...
         uses: actions/setup-go@v2
         with:
           go-version: 1.13.3
         id: go
-      - name: Install linters
+      - name: Install buildifier
         run: go get github.com/bazelbuild/buildtools/buildifier
       - uses: actions/checkout@v2
-      - name: C++ Build
-        run: .github/workflows/scripts/build_cpp.sh
```
Comment (bcebere) [inline .github/workflows/tests.yml:32]: The build_cpp step was added for preventing warnings at build time, such as size_t vs int comparisons etc.

I didn't want to enforce that on every platform, that's why it was I separate step. 
Is SEAL generating warnings now on Ubuntu?

**Review by bcebere — APPROVED** [review]
Not sure about the build_cpp removal, otherwise LGTM.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (s0l0ist) [reply .github/workflows/tests.yml:32]: @bcebere In the test workflow it doesn't show warnings. Is that what you mean?

Comment (bcebere) [reply .github/workflows/tests.yml:32]: It was supposed to fail on warnings at build time, it had the -Werrror flag in addition to the regular build.

Comment (s0l0ist) [reply .github/workflows/tests.yml:32]: After discussing internally, we've come to the conclusion it is safe to merge without the above build script.

**Pull request merged**